From 57673c2b0baa900dddae3b9eb3d7748ebf550eb3 Mon Sep 17 00:00:00 2001
From: Rusty Russell <rusty@rustcorp.com.au>
Date: Mon, 31 Mar 2014 14:39:57 +1030
Subject: [PATCH 2/2] Use 'E' instead of 'X' for unsigned module taint flag.

Git-commit: 57673c2b0baa900dddae3b9eb3d7748ebf550eb3
Patch-mainline: 3.15-rc?, merged to Rusty Russell's git tree
References: bnc#870450
Target: SLE-12

Takashi Iwai <tiwai@suse.de> says:
> The letter 'X' has been already used for SUSE kernels for very long
> time, to indicate the external supported modules.  Can the new flag be
> changed to another letter for avoiding conflict...?
> (BTW, we also use 'N' for "no support", too.)

Note: this code should be cleaned up, so we don't have such maps in
three places!

Signed-off-by: Rusty Russell <rusty@rustcorp.com.au>
Acked-by: Lee, Chun-Yi <jlee@suse.com>

Index: linux-3.12-SLE12/Documentation/ABI/testing/sysfs-module
===================================================================
--- linux-3.12-SLE12.orig/Documentation/ABI/testing/sysfs-module
+++ linux-3.12-SLE12/Documentation/ABI/testing/sysfs-module
@@ -49,4 +49,4 @@ Description:	Module taint flags:
 			O - out-of-tree module
 			F - force-loaded module
 			C - staging driver module
-			X - unsigned module
+			E - unsigned module
Index: linux-3.12-SLE12/Documentation/oops-tracing.txt
===================================================================
--- linux-3.12-SLE12.orig/Documentation/oops-tracing.txt
+++ linux-3.12-SLE12/Documentation/oops-tracing.txt
@@ -265,7 +265,7 @@ characters, each representing a particul
 
  13: 'O' if an externally-built ("out-of-tree") module has been loaded.
 
- 14: 'X' if an unsigned module has been loaded in a kernel supporting
+ 14: 'E' if an unsigned module has been loaded in a kernel supporting
      module signature.
 
 The primary reason for the 'Tainted: ' string is to tell kernel
Index: linux-3.12-SLE12/kernel/module.c
===================================================================
--- linux-3.12-SLE12.orig/kernel/module.c
+++ linux-3.12-SLE12/kernel/module.c
@@ -1065,7 +1065,7 @@ static size_t module_flags_taint(struct
 	if (mod->taints & (1 << TAINT_CRAP))
 		buf[l++] = 'C';
 	if (mod->taints & (1 << TAINT_UNSIGNED_MODULE))
-		buf[l++] = 'X';
+		buf[l++] = 'E';
 #ifdef CONFIG_SUSE_KERNEL_SUPPORTED
 	if (mod->taints & (1 << TAINT_NO_SUPPORT))
 		buf[l++] = 'N';
Index: linux-3.12-SLE12/kernel/panic.c
===================================================================
--- linux-3.12-SLE12.orig/kernel/panic.c
+++ linux-3.12-SLE12/kernel/panic.c
@@ -210,7 +210,7 @@ static const struct tnt tnts[] = {
 	{ TAINT_CRAP,			'C', ' ' },
 	{ TAINT_FIRMWARE_WORKAROUND,	'I', ' ' },
 	{ TAINT_OOT_MODULE,		'O', ' ' },
-	{ TAINT_UNSIGNED_MODULE,        'X', ' ' },
+	{ TAINT_UNSIGNED_MODULE,        'E', ' ' },
 #ifdef CONFIG_SUSE_KERNEL_SUPPORTED
 	{ TAINT_NO_SUPPORT,		'N', ' ' },
 	{ TAINT_EXTERNAL_SUPPORT,	'X', ' ' },
@@ -233,7 +233,7 @@ static const struct tnt tnts[] = {
  *  'C' - modules from drivers/staging are loaded.
  *  'I' - Working around severe firmware bug.
  *  'O' - Out-of-tree module has been loaded.
- *  'X' - Unsigned module has been loaded.
+ *  'E' - Unsigned module has been loaded.
  *  'N' - Unsuported modules loaded.
  *  'X' - Modules with external support loaded.
  *
